pkgs/development/compilers/sbcl/default.nix

```diff
@@ -94,7 +94,14 @@ stdenv.mkDerivation (self: rec {
       strace
     ]
   );
-  buildInputs = lib.optionals coreCompression [ zstd ];
+  buildInputs = lib.optionals coreCompression (
+    # Declare at the point of actual use in case the caller wants to override
+    # buildInputs to sidestep this.
+    assert lib.assertMsg (!purgeNixReferences) ''
```

Comment:
maybe this should be a warning and not an assert?

Reply:
As you prefer / as is simpler — if we use dynamic `zstd` here and thus defeat the entire point of avoiding Nix references, I think it is fine to fail.

Reply:
I've thought about it and I'm going to leave it an assert, because it's so easy to overrule, and so unlikely to get hit accidentally. It's worth it to fail loudly here and force the user to explicitly address the situation so they don't get caught off guard.